### PR TITLE
Refactor theme objects

### PR DESCRIPTION
**Pull request by live627**
Kicking off 2.0-dev...

### CODE REVIEW

File: themes/default/Theme.php [hunk 1]
```diff
@@ -24,9 +26,104 @@
...
  */
-class Theme extends \Theme
+class Theme extends BaseTheme
 {
 	/**
+	 * Initialize the template... mainly little settings.
+	 *
+	 * @return array Theme settings
+	 */
+	function getSetings()
```
Comment (emanuele45) [inline themes/default/Theme.php]: Setings => Settings? :innocent:

**Comment by emanuele45** [comment]
I was just having a second look at this PR: @live627 what would you think to call theme() only once when the controller is instantiated (something like the HttpReq class):
```
		$this->_req = HttpReq::instance();
		$this->_theme = theme();
```
that would save some function calls and prepare for a day we will pass the instance to the controller as argument?

**Comment by live627** [comment]
I'm finished with this one now.

File: sources/subs/Themes/TemplateLayers.php [hunk 2]
```diff
@@ -184,33 +211,24 @@ public function isError()
...
 	 *
+	 * @deprecated use the theme object
+	 *
 	 * @param boolean $error_safe if error mode is on or off
-	 * @return Template_Layers instance of the class
+	 *
+	 * @return ElkArte\Theme\TemplateLayers instance of the class
 	 */
-	public static function instance($error_safe = false)
+	public static function getInstance($error_safe = false)
```
Comment (emanuele45) [inline sources/subs/Themes/TemplateLayers.php]: Preferred name is ```instance```. For consistency with all the other singleton. :innocent:

**Comment by live627** [comment]
This will conflict with 4a2269473f8222b04cc714a33d3b5994ab52ee6f

**Comment by emanuele45** [comment]
No problem, I'll change mine after this is merged. :wink:

**Comment by live627** [comment]
Is this going to be merged soon?

**Comment by emanuele45** [comment]
I was considering what was less painful: wait a bit to have 1.1.1 changes in dev and then let you deal with possible conflicts or merge this and have to deal with possible conflicts myself later when merging.

A bit of conflict of interests! :laughing: 
I guess I can merge that and deal with 1.1.1 later.

**Comment by emanuele45** [comment]
Oh well, apparently it's still able to merge without issues, so bring it in! :D

**Comment by live627** [comment]
> I was considering what was less painful: wait a bit to have 1.1.1 changes in dev and then let you deal with possible conflicts or merge this and have to deal with possible conflicts myself later when merging.

I just checked and there are no conflicts. I also checked the patch branch by @Spuds with same result (no conflicts).

**Pull request merged**